tests/next_tests/integration_tests/feature_tests/ffront_tests/test_compiled_program.py

```diff
@@ -0,0 +1,693 @@
+    args, kwargs = cases.get_default_data(unstructured_case, compile_testee_unstructured)
+
+    # make sure the backend is never called
+    object.__setattr__(compile_testee_unstructured, "backend", _always_raise_callable)
+
+    with pytest.raises(RuntimeError, match="No program.*static.*arg.*"):
+        compile_testee_unstructured(
+            *args, offset_provider=unstructured_case.offset_provider, **kwargs
+        )
+
+
+def test_compile_unstructured_modified_offset_provider(
+    unstructured_case, compile_testee_unstructured, skip_value_mesh_descriptor
+):
+    if unstructured_case.backend is None:
+        pytest.skip("Embedded compiled program doesn't make sense.")
+    object.__setattr__(compile_testee_unstructured, "enable_jit", False)
+
+    # compiled for skip_value_mesh
+    compile_testee_unstructured.compile(
+        offset_provider_type=skip_value_mesh_descriptor.offset_provider,
+    )
+
+    # but executing the simple_mesh
+    args, kwargs = cases.get_default_data(unstructured_case, compile_testee_unstructured)
+
+    # make sure the backend is never called
+    object.__setattr__(compile_testee_unstructured, "backend", _always_raise_callable)
+
+    with pytest.raises(RuntimeError, match="No program.*static.*arg.*"):
+        compile_testee_unstructured(
+            *args, offset_provider=unstructured_case.offset_provider, **kwargs
+        )
+
+
+@pytest.fixture
+def compile_variants_field_operator():
+    @gtx.field_operator
+    def compile_variants_field_operator(
+        field_a: cases.IField,
+        scalar_int: int32,
+        scalar_float: float,
+        scalar_bool: bool,
+        field_b: cases.IFloatField,
+    ) -> tuple[cases.IField, cases.IFloatField]:
+        return (
+            (field_a + scalar_int, field_b + scalar_float)
+            if scalar_bool
+            else (field_a - scalar_int, field_b - scalar_float)
+        )
+
+    return compile_variants_field_operator
+
+
+@pytest.fixture
+def compile_variants_testee_not_compiled(
+    cartesian_case, compile_variants_field_operator
+) -> Program:
+    if cartesian_case.backend is None:
+        pytest.skip("Embedded compiled program doesn't make sense.")
+
+    @gtx.program(backend=cartesian_case.backend)
+    def testee(
+        field_a: cases.IField,
+        scalar_int: int32,
+        scalar_float: float,
+        scalar_bool: bool,
+        field_b: cases.IFloatField,
+        out: tuple[cases.IField, cases.IFloatField],
+    ):
+        compile_variants_field_operator(
+            field_a, scalar_int, scalar_float, scalar_bool, field_b, out=out
+        )
+
+    return testee
+
+
+@pytest.fixture
+def compile_variants_testee(cartesian_case, compile_variants_testee_not_compiled) -> Program:
+    return compile_variants_testee_not_compiled.compile(
+        scalar_int=[1, 2],
+        scalar_float=[3.0, 4.0],
+        scalar_bool=[True, False],
+        offset_provider_type=cartesian_case.offset_provider,
```

Comment:
for consistency this needs to be
```suggestion
        offset_provider_type=[cartesian_case.offset_provider],
```
and a test should be added for compiling for 2 offset_providers...